docs/examples/image_segmentation/region_growing.jl

```diff
@@ -0,0 +1,120 @@
+# ---
+# cover: assets/regiongrowing.gif
+# title: Region Growing Algorithm
+# description: This demo shows how to use the region growing algorithm to segment an image.
+# author: Ashwani Rathee
+# date: 2021-08-25
+# ---
+
+# Region Growing Algorithm is a procedure that groups pixels or sub regions
+# into a larger regions. We use principle of similiarity to group similar pixels.
+# Principle of similiarity states that a region is coherent if all the pixels of that
+# region are homogeneous and criterion of homogenity could be glcm, color, texture
+# shape and model etc
+
+# Major steps in a region growing algorithm:
+# - Selection of initial seed
+# - Seed Growing Criteria
+# - Termination of the segmentation process
+
+# In this demonstration, we will segment an image using the region growing algorithm.
+# There are two methods to do region growing - the first one is the seeded region growing method,
+# second one is the unseeded region growing method.
+
+using Images
+using ImageSegmentation
+using Random
+
+function get_random_color(seed)
+    Random.seed!(seed)
+    rand(RGB{N0f8})
+end
+
+# ## Seeded Region Growing
+
+# Seeded region growing segments an image with respect to some user-defined seeds.
+# Each seed is a `(position, label)`` tuple, where `position` is a `CartesianIndex` and
+# label is a positive integer. Each label corresponds to a unique partition of 
+# the image. The algorithm tries to assign these labels to each of the remaining 
+# points. If more than one point has the same label then they will be contribute
+# to the same segment.
+
+img = load("assets/worm.jpg")
```

Comment:
Generally, we should not expect users have an "assets" folder at hand. We should either add it to TestImages or using some explicit `download` tricks.

Reply:
I think I'll add worm.png to imgur then